### ISSUE TITLE
Performance issues,

### ISSUE THREAD

**Issue by morenol** [wasm-tool/rollup-plugin-rust#38]
It looks like this only uses one cpu core.

```
/home/user/.nvm/versions/node/v14.18.0/bin/node /home/user/Sources/repo/node_modules/.bin/wasm-opt index_bg.wasm --output wasm_opt.wasm -O4
```

Using `wasm-opt` binary from (`apt-get install binaryen`) makes it more faster.

**Comment by Pauan** [comment]
I'm confused, this already [uses `wasm-opt` from binaryen](https://github.com/wasm-tool/rollup-plugin-rust/blob/6ad48be85acbada1e5accad5c9185e9bae819b88/package.json#L31). Unless you're saying there's a difference between using the npm package or the distro's built-in package?

We can't really use the distro's package because that's too much of a hassle for users, but it would be possible to put in an option to specify a custom binary.

**Comment by nacardin** [comment]
Yes, there is a difference. Looks like the `binaryen` package (https://github.com/AssemblyScript/binaryen.js) is transpiling the tools into javascript. This can only run single threaded, which is not optimal. An option to specify the binary path using an env variable may be enough.

**Comment by Pauan** [comment]
I like the idea of an env var, since it's such a niche thing.

**Comment by Pauan** [comment]
I just published version 2.3.2 which has [three new env vars](https://github.com/wasm-tool/rollup-plugin-rust#environment-variables).

You should be able to use `env WASM_OPT_BIN="$(which wasm-opt)" yarn build` to use the system wasm-opt.

**Comment by Pauan** [comment]
By the way, you might want to report the single core issue to binaryen, since Node can use multiple threads, so they should probably fix that upstream.

**Comment by nacardin** [comment]
> I just published version 2.3.2 which has [three new env vars](https://github.com/wasm-tool/rollup-plugin-rust#environment-variables).
> 
> You should be able to use `env WASM_OPT_BIN="$(which wasm-opt)" yarn build` to use the system wasm-opt.

Thanks! This solved my issue.
